leap/leap.go

```diff
@@ -0,0 +1,23 @@
+package main
+
+import "fmt"
+
+func main() {
+	fmt.Println("The program determines if a year is a leap year.")
+	fmt.Print("Enter the year: ")
+	var year int
+	fmt.Scan(&year)
+	if year%100 == 0 {
```

Comment:
Please try to create a more complicated conditions such that the code would look like this:
```
if COMPLICATED_CONDITION {
  fmt.Println(year, "is a leap year")
} else {
  fmt.Println(year, "is not a leap year")
}
```